File: src/main/java/me/krzu/hsbmechanics/listeners/CraftListeners.java
package me.krzu.hsbmechanics.listeners;

import me.krzu.hsbmechanics.HSBMechanics;
import me.krzu.hsbmechanics.inventories.crafting.CraftMenu;
import org.bukkit.Material;
import org.bukkit.block.Block;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.block.Action;
import org.bukkit.event.player.PlayerInteractEvent;

public class CraftListeners implements Listener {

    private final HSBMechanics plugin;

    public CraftListeners(HSBMechanics plugin) {
        this.plugin = plugin;
    }

    @EventHandler
    public void onPlayerInteract (PlayerInteractEvent event) {
        Player player = event.getPlayer();
        Action action = event.getAction();

        if (action.isRightClick()) {
            if (player.isSneaking() && event.isBlockInHand()) {
                return;
            }

            Block block = event.getClickedBlock();

            if (block != null && block.getType() == Material.CRAFTING_TABLE) {
                event.setCancelled(true);

                new CraftMenu(plugin)
                        .build(player);
            }
        }
    }
}

File: src/main/kotlin/com/amniel/hsbmechanics/listeners/CraftListeners.kt
package com.amniel.hsbmechanics.listeners

import com.amniel.hsbmechanics.HSBMechanics
import com.amniel.hsbmechanics.inventories.crafting.CraftMenu
import org.bukkit.Material
import org.bukkit.event.Listener
import org.bukkit.event.EventHandler
import org.bukkit.event.player.PlayerInteractEvent

class CraftListeners(val plugin: HSBMechanics) : Listener {

    @EventHandler
    fun onPlayerInteract(event: PlayerInteractEvent) {
        val player = event.player
        val action = event.action

        if (action.isRightClick) {
            if (player.isSneaking && event.isBlockInHand) {
                return
            }
            val block = event.clickedBlock

            if (block != null && block.type == Material.CRAFTING_TABLE) {
                event.isCancelled = true
                CraftMenu(plugin).build(player)
            }
        }
    }
}